qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -73,6 +73,17 @@ def init():
     download_manager.install(webenginesettings.private_profile)
     objreg.register('webengine-download-manager', download_manager)
 
+    # Clear visited links on web history clear
+    hist = objreg.get('web-history')
+    hist.history_cleared.connect(
+        webenginesettings.default_profile.clearAllVisitedLinks)
+    hist.history_cleared.connect(
+        webenginesettings.private_profile.clearAllVisitedLinks)
+    hist.url_cleared.connect(
+        lambda url: webenginesettings.default_profile.clearVisitedLinks([url]))
+    hist.url_cleared.connect(
+        lambda url: webenginesettings.private_profile.clearVisitedLinks([url]))
```

Comment:
No strong feelings on this one, but might be worth making it a for-loop maybe? (`for profile in [webenginesettings.default_profile, webenginesettings.private_profile]: ...`)